File: LearningLens2025/legacy_files/analytics/focused_ai_ui/lib/models/CourseWorkWithSubmissions.dart
import 'package:focused_ai_ui/models/grade.dart';

class CourseWorkWithSubmissions {
  final String courseWorkId;
  final String title;
  final String workType;
  final List<Grade> grades;

  CourseWorkWithSubmissions({
    required this.courseWorkId,
    required this.title,
    required this.workType,
    required this.grades,
  });

  factory CourseWorkWithSubmissions.fromJson(Map<String, dynamic> json) {
    return CourseWorkWithSubmissions(
      courseWorkId: json['courseWorkId'],
      title: json['title'],
      workType: json['workType'],
      grades: (json['grades'] as List).map((g) => Grade.fromJson(g)).toList(),
    );
  }
}

File: LearningLens2025/legacy_files/focused-ai-analytics-service/focused_ai_services/src/main/java/com/focused_ai/models/domain/CourseWorkWithSubmissions.java
package com.focused_ai.models.domain;

import java.util.List;

public class CourseWorkWithSubmissions {
    private String courseWorkId;
    private String title;
    private String workType;
    private List<Grade> grades;

    public CourseWorkWithSubmissions(String courseWorkId, String title, String workType, List<Grade> grades) {
        this.courseWorkId = courseWorkId;
        this.title = title;
        this.workType = workType;
        this.grades = grades;
    }

    public String getCourseWorkId() { return courseWorkId; }
    public String getTitle() { return title; }
    public String getWorkType() { return workType; }
    public List<Grade> getGrades() { return grades; }
}
